domains.go

```diff
@@ -74,3 +77,21 @@ func (c *Client) DomainsShow(app string, id string) (Domain, error) {
 
 	return domainRes.Domain, nil
 }
+
+func (c *Client) DomainSetCanonical(app, id string) (Domain, error) {
+	var domainRes DomainRes
+	err := c.subresourceUpdate(app, "domains", id, DomainRes{Domain: Domain{Canonical: true}}, &domainRes)
+	if err != nil {
+		return Domain{}, errgo.Mask(err)
+	}
+	return domainRes.Domain, nil
+}
+
+func (c *Client) DomainUnsetCanonical(app, id string) (Domain, error) {
```

Comment:
Why would this method need an id ? There can be only one canonical domain per app.